### PR TITLE
Add a dispatch context and provider

### PR DESCRIPTION
**Pull request by arkadyan**

### CODE REVIEW

File: assets/src/contexts/dispatchContext.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { createContext } from "react"
+import { Dispatch } from "../state"
+
+// tslint:disable-next-line: no-empty
+const noopDispatch: Dispatch = () => {}
```
Comment (skyqrose) [inline assets/src/contexts/dispatchContext.ts]: Why is this needed?

Comment (arkadyan) [reply assets/src/contexts/dispatchContext.ts]: `createContext` needs a default value. My reading of the documentation is that this is mostly to support testing, and that this default value is never used when you are using a provider around your component.
